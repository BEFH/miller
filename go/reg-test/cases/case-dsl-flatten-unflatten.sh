run_cat $indir/flatten-input-1.json

run_mlr --ijson --oxtab --from $indir/flatten-input-1.json put '
  map o = {};
  for (k, v in $*) {
    for (k2, v2 in flatten(k, ".", v)) {
      o[k2] = v2
    }
  }
  $* = o;
'

run_mlr --ijson --oxtab --from $indir/flatten-input-1.json put '$* = flatten("", ".", $*)'

run_mlr --ijson --oxtab --from $indir/flatten-input-1.json put '$* = flatten($*, ".")'

run_mlr --ijson --ojson --no-auto-unflatten --from $indir/flatten-input-1.json put '$a = flatten("a", ".", $a)'

run_mlr --ijson --ojson --no-auto-unflatten --from $indir/flatten-input-1.json put '$b = flatten("b", ".", $b)'

run_mlr --ijson --oxtab --from $indir/flatten-input-2.json put '$* = flatten("", ".", $*)'

run_mlr --ixtab --ojson --no-auto-unflatten --from $indir/unflatten-input.xtab put '$* = unflatten($*, ".")'
