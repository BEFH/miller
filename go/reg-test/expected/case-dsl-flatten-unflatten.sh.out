
cat ./reg-test/input/flatten-input-1.json
{
  "a": 1,
  "b": {
    "c": 2,
    "d": 3
  }
}
{
  "a": 1,
  "b": {
    "c": {
      "x": 2,
      "y": 3
    },
    "d": {
      "x": 4,
      "y": 5
    }
  }
}
{
  "a": 1,
  "b": {
    "c": [
      2,
      3
    ]
  }
}
{
  "a": 1,
  "b": [
    {
      "x": 2,
      "y": 3
    },
    {
      "x": 4,
      "y": 5
    }
  ]
}

mlr --ijson --oxtab --from ./reg-test/input/flatten-input-1.json put 
  map o = {};
  for (k, v in $*) {
    for (k2, v2 in flatten(k, ".", v)) {
      o[k2] = v2
    }
  }
  $* = o;

b.c 2
b.d 3

b.c.x 2
b.c.y 3
b.d.x 4
b.d.y 5

b.c.1 2
b.c.2 3

b.1.x 2
b.1.y 3
b.2.x 4
b.2.y 5

mlr --ijson --oxtab --from ./reg-test/input/flatten-input-1.json put $* = flatten("", ".", $*)
a   1
b.c 2
b.d 3

a     1
b.c.x 2
b.c.y 3
b.d.x 4
b.d.y 5

a     1
b.c.1 2
b.c.2 3

a     1
b.1.x 2
b.1.y 3
b.2.x 4
b.2.y 5

mlr --ijson --oxtab --from ./reg-test/input/flatten-input-1.json put $* = flatten($*, ".")
a   1
b.c 2
b.d 3

a     1
b.c.x 2
b.c.y 3
b.d.x 4
b.d.y 5

a     1
b.c.1 2
b.c.2 3

a     1
b.1.x 2
b.1.y 3
b.2.x 4
b.2.y 5

mlr --ijson --ojson --no-auto-unflatten --from ./reg-test/input/flatten-input-1.json put $a = flatten("a", ".", $a)
{
  "a": 1,
  "b": {
    "c": 2,
    "d": 3
  }
}
{
  "a": 1,
  "b": {
    "c": {
      "x": 2,
      "y": 3
    },
    "d": {
      "x": 4,
      "y": 5
    }
  }
}
{
  "a": 1,
  "b": {
    "c": [2, 3]
  }
}
{
  "a": 1,
  "b": [
    {
      "x": 2,
      "y": 3
    },
    {
      "x": 4,
      "y": 5
    }
  ]
}

mlr --ijson --ojson --no-auto-unflatten --from ./reg-test/input/flatten-input-1.json put $b = flatten("b", ".", $b)
{
  "a": 1,
  "b": {
    "b.c": 2,
    "b.d": 3
  }
}
{
  "a": 1,
  "b": {
    "b.c.x": 2,
    "b.c.y": 3,
    "b.d.x": 4,
    "b.d.y": 5
  }
}
{
  "a": 1,
  "b": {
    "b.c.1": 2,
    "b.c.2": 3
  }
}
{
  "a": 1,
  "b": {
    "b.1.x": 2,
    "b.1.y": 3,
    "b.2.x": 4,
    "b.2.y": 5
  }
}

mlr --ijson --oxtab --from ./reg-test/input/flatten-input-2.json put $* = flatten("", ".", $*)
hostname                    localhost
pid                         12345
req.id                      6789
req.method                  GET
req.path                    api/check
req.host                    foo.bar
req.headers.host            bar.baz
req.headers.user-agent      browser
res.status_code             200
res.header.content-type     text
res.header.content-encoding plain
empty1                      {}
empty2                      []
wrapper.empty3              {}
wrapper.emtpy4              []

mlr --ixtab --ojson --no-auto-unflatten --from ./reg-test/input/unflatten-input.xtab put $* = unflatten($*, ".")
{
  "hostname": "localhost",
  "pid": 12345,
  "req": {
    "id": 6789,
    "method": "GET",
    "path": "api/check",
    "host": "foo.bar",
    "headers": {
      "host": "bar.baz",
      "user-agent": "browser"
    }
  },
  "res": {
    "status_code": 200,
    "header": {
      "content-type": "text",
      "content-encoding": "plain"
    }
  },
  "empty1": "{}",
  "empty2": "[]",
  "wrapper": {
    "empty3": "{}",
    "emtpy4": "[]"
  }
}

